### PR TITLE
Azure Blob Storage STAC provider

### PR DESCRIPTION
**Pull request by tomkralidis**
# Overview
This PR adds functionality to expose Microsoft Blob Storage as a STAC catalog.

# Related Issue / Discussion
None
# Additional Information
None
# Contributions and Licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [x] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution.
- [x] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

**Review by kalxas — APPROVED** [review]

File: requirements-provider.txt
Comment (webb-ben) [inline requirements-provider.txt:1]: Do these installs also need to be included in requirements-docker.txt? I don't see debian distributions for them

File: pygeoapi/provider/azure_.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# =================================================================
+
+from datetime import datetime
+import logging
+import os
+from urllib.parse import urljoin
```
Comment (webb-ben) [inline pygeoapi/provider/azure_.py:33]: use pygeoapi.util url_join?

**Review by webb-ben — COMMENTED** [review]
Is GitHub Action testing possible (and/or required) for this provider? I don't see any tests for STAC providers...

File: requirements-provider.txt
Comment (tomkralidis) [reply requirements-provider.txt:1]: > Is GitHub Action testing possible (and/or required) for this provider? I don't see any tests for STAC providers...

We have tests for the [FileSystemProvider](https://github.com/geopython/pygeoapi/blob/master/tests/test_filesystem_provider.py).  Adding tests for Azure would mean setting up a local Azure instance (azurite) in CI.  If there is enough use/demand on STAC/Azure we can add in the future.

Comment (tomkralidis) [reply requirements-provider.txt:1]: > 

If you mean the azure dependencies, they are best placed in `requirements-provider.txt`.

File: pygeoapi/provider/azure_.py (hunk 1 above)
Comment (tomkralidis) [reply pygeoapi/provider/azure_.py]: updated.

File: requirements-provider.txt
Comment (webb-ben) [reply requirements-provider.txt:1]: Agree that it _should_ be in `requirements-provider.txt`. 

In practice, `requirements-docker.txt` installs the packages that are in `requirements-provider.txt`, but not available as debian distro.

Should https://github.com/geopython/pygeoapi/blob/9debd89cd832e4155d335001376a49cd4f4755c3/Dockerfile#L125
be 
```
&& pip3 install -r  requirements-provider.txt \
```
?

**Pull request merged**